File: scalechain-util/src/main/scala/io/scalechain/util/GlobalStopWatch.scala
package io.scalechain.util

/**
  * Created by kangmo on 7/10/16.
  */
object GlobalStopWatch extends StopWatchEx {}


class StopWatchEx extends StopWatch {
  def measure[T] (subject : String)(block : => T): T = {
    val watchSubject : StopWatchSubject = start(subject)
    val returnValue = block
    watchSubject.stop()
    returnValue
  }
}

File: scalechain-util/src/main/kotlin/io/scalechain/util/GlobalStopWatch.kt
package io.scalechain.util


open class StopWatchEx : StopWatch() {
    fun <T> measure (subject : String, block : () -> T): T  {
        val watchSubject : StopWatchSubject = start(subject)
        val returnValue = block()
        watchSubject.stop()
        return returnValue
    }
}

/**
 * Created by kangmo on 7/10/16.
 */
object GlobalStopWatch : StopWatchEx()



